source/Halibut/HalibutRuntimeBuilder.cs

```diff
@@ -106,6 +107,12 @@ internal HalibutRuntimeBuilder WithPollingReconnectRetryPolicy(Func<RetryPolicy>
             return this;
         }
 
+        internal HalibutRuntimeBuilder WithControlMessageObserver(IControlMessageObserver controlMessageObserver)
```

Comment:
internal since ideally clients would not call this (if they ever did we would have no hesitation to break compatibility ie we wont support this observer outside of our testing needs)